test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -83,31 +87,56 @@ public void EnlistOff()
             using (var conn2 = OpenConnection(ConnectionStringEnlistOff))
             {
                 Assert.That(conn1.EnlistedTransaction, Is.Null);
-                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1));
-                Assert.That(conn2.ExecuteScalar("SELECT COUNT(*) FROM data"), Is.EqualTo(1));
+                Assert.That(conn1.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1),
+                    "Unexpected insert rowcount");
+                Assert.That(conn2.ExecuteScalar("SELECT COUNT(*) FROM data"), Is.EqualTo(1),
+                    "Unexpected data count");
+            }
+
+            // Scope disposed and not completed => rollback, but no enlistment, so changes should still be there.
+            using (var conn3 = OpenConnection(ConnectionStringEnlistOff))
+            {
+                Assert.That(conn3.ExecuteScalar("SELECT COUNT(*) FROM data"), Is.EqualTo(1), "Insert unexpectedly rollback-ed");
             }
         }
 
-        [Test, Description("Single connection rollback")]
-        public void Rollback()
+        [Test, Description("Single connection enlisting explicitly, rollback")]
+        public void RollbackExplicitEnlist()
         {
             using (var conn = OpenConnection())
             {
                 using (new TransactionScope())
                 {
                     conn.EnlistTransaction(Transaction.Current);
-                    Assert.That(conn.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1));
+                    Assert.That(conn.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1),
+                        "Unexpected insert rowcount");
                     // No commit
                 }
                 AssertNoPreparedTransactions();
                 using (var tx = conn.BeginTransaction())
                 {
-                    Assert.That(conn.ExecuteScalar(@"SELECT COUNT(*) FROM data"), Is.EqualTo(0));
+                    Assert.That(conn.ExecuteScalar(@"SELECT COUNT(*) FROM data"), Is.EqualTo(0),
+                        "Unexpected data count");
                     tx.Rollback();
                 }
             }
         }
 
+        [Test, Description("Single connection enlisting implicitly, rollback")]
+        public void RollbackImplicitEnlist()
+        {
+            using (new TransactionScope())
+            using (var conn = OpenConnection(ConnectionStringEnlistOn))
+            {
+                Assert.That(conn.ExecuteNonQuery(@"INSERT INTO data (name) VALUES ('test')"), Is.EqualTo(1),
+                    "Unexpected insert rowcount");
+                AssertNoPreparedTransactions();
+                // No commit
+            }
+
+            AssertNumberOfRows(0);
```

Comment:
Like `Enlist=false` but the other way round, checking the rollback case with `Enlist=true` seems to me the best way to ascertain the enlistment.